src/consumers/common.js

```diff
@@ -2,9 +2,21 @@ const Promise = require("bluebird");
 
 const { EXHAUSTED } = require("../sequence");
 
-
-const INCOMPLETE = Symbol();
-
+function next (renderer, iter, push) {
+  const max = renderer.batchSize;
+  const nextVal = renderer._next();
+
+  if (nextVal === EXHAUSTED) {
+    return EXHAUSTED;
+  } else if (nextVal instanceof Promise) {
+    return nextVal
+    .then(push)
+    .then(() => iter < max && next(renderer, iter + 1, push));
+  } else {
+    push(nextVal);
+    return Promise.resolve(nextVal);
+  }
+}
```

Comment:
Could probably find a more elegant way to do this iteration. The crux of the issue is that the previous promise has to resolve before we can invoke kick off the next promise. Any other solution requires a new library or babel preset and I wasn't sure if that would be desirable. If anyone knows how to do this with `bluebird` please let me know!

Some other options:
- https://caolan.github.io/async/docs.html#series
- https://github.com/dbushong/async-q#series
- ES6 generators